roles/deploy/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Initialize
+  deploy_helper: path="{{ project_root }}" state=present
+
+- name: Clone project files
+  git: repo="{{ project_git_repo }}"
+       dest="{{ project_source_path }}"
+       version="{{ project_version }}"
+       accept_hostkey=yes
+
+- name: write unfinished file
+  file: path="{{ project_source_path }}/{{ deploy_helper.unfinished_filename }}" state=touch
+
+- name: Copy files to new build dir
+  command: cp -pr {{ project_source_path }} {{ deploy_helper.new_release_path }}
```

Comment:
I asked 
>could you just clone the project straight into `deploy_helper.new_release_path` instead of cloning to one dir then copying to another?

I now see the benefit of idempotence in cloning into `project_source_path` (`example.com/shared/source`). It will skip the cloning task if it already has the specified version. If cloning straight into the new release path as I proposed, it would perform the cloning every time.